lensless/rfft_convolve.py

```diff
@@ -68,50 +68,68 @@ def __init__(self, psf, dtype=None, pad=True, norm="ortho", **kwargs):
                 self._pad(self._psf), norm=norm, dim=(0, 1), s=self._padded_shape[:2]
             )
             self._Hadj = torch.conj(self._H)
-            self._padded_data = torch.zeros(size=self._padded_shape, dtype=dtype, device=psf.device)
-
+            # self._padded_data = torch.zeros(size=self._padded_shape, dtype=dtype, device=psf.device)
```

Comment:
why commented? otherwise `self._padded_data` is not defined / allocated for torch

Reply:
To allow for gradient calculation, we cannot reuse the same tensor. It is now defined when needed.